packages/api/src/external/fhir/document/index.ts

```diff
@@ -30,14 +33,24 @@ import { metriportDataSourceExtension } from "../shared/extensions/metriport";
 dayjs.extend(isToday);
 
 export const MAX_FHIR_DOC_ID_LENGTH = 64;
-export const authorTypes = [
-  "Device",
-  "Organization",
-  "Patient",
-  "Practitioner",
-  "PractitionerRole",
-  "RelatedPerson",
-];
+
+type AuthorTypes =
+  | Device
+  | Organization
+  | PatientFHIR
+  | Practitioner
+  | PractitionerRole
+  | RelatedPerson;
+const authorTypesMap: Record<AuthorTypes["resourceType"], AuthorTypes["resourceType"]> = {
+  Device: "Device",
+  Organization: "Organization",
+  Patient: "Patient",
+  Practitioner: "Practitioner",
+  PractitionerRole: "PractitionerRole",
+  RelatedPerson: "RelatedPerson",
+};
```

Comment:
Painful but the only way I could find to have compile-time errors in case someone includes the wrong type/value.